implementation/test/DepositUtilsTest.js

```diff
@@ -376,7 +382,61 @@ describe("DepositUtils", async function() {
   })
 
   describe("auctionValue()", async () => {
-    it.skip("is TODO")
+    let duration
+    let basePercentage
+    before(async () => {
+      duration = await tbtcConstants.getAuctionDuration.call()
+      basePercentage = await tbtcConstants.getAuctionBasePercentage.call()
+      auctionValue = new BN(100000000)
+    })
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("returns base value if no time has elapsed", async () => {
+      testDeposit.send(auctionValue, {from: accounts[0]})
+      const block = await web3.eth.getBlock("latest")
+
+      await testDeposit.setLiquidationAndCourtesyInitated(block.timestamp, 0)
+
+      const value = await testDeposit.auctionValue.call()
+      expect(value).to.eq.BN(auctionValue.mul(basePercentage).div(new BN(100)))
+    })
+
+    it("returns full value if auction Duration has elapsed ", async () => {
+      testDeposit.send(auctionValue, {from: accounts[0]})
+      const block = await web3.eth.getBlock("latest")
+
+      await testDeposit.setLiquidationAndCourtesyInitated(block.timestamp, 0)
+      await increaseTime(duration.toNumber())
+
+      const value = await testDeposit.auctionValue.call()
+      expect(value).to.eq.BN(auctionValue)
+    })
+
+    it("scales auction value currectly", async () => {
+      testDeposit.send(auctionValue, {from: accounts[0]})
+      const block = await web3.eth.getBlock("latest")
+
+      await testDeposit.setLiquidationAndCourtesyInitated(block.timestamp, 0)
+
+      const elapsedTime = duration.div(new BN(2))
+      const elapsedPercent = new BN(100)
+        .sub(basePercentage)
+        .mul(elapsedTime)
+        .div(duration)
+      const percentage = basePercentage.add(elapsedPercent)
+
+      // elapse half the auction time
+      await increaseTime(elapsedTime.toNumber())
+
+      const value = await testDeposit.auctionValue.call()
+      expect(value).to.eq.BN(auctionValue.mul(percentage).div(new BN(100)))
```

Comment:
I wonder if it would make sense to make this some multi-value test, e.g. 10 random increments from 0 to 100, testing that things are right along the way.

Reply:
Maybe. I'm not sure if the benefit is worth the extra tests. Testing for half-value at half-time is a cheap way to test assuming linear scaling.

Reply:
In general, yes, but making a habit of multi-value tests is long-term more beneficial.

That said, no need to block on it.